### PR TITLE
OfficeRequest, OfficeDetails, OfficeCreator,  tests

### PR DESCRIPTION
**Pull request by Artsapollo**

### CODE REVIEW

File: src/main/java/springData/dto/OfficeRequest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private BigDecimal office;
+
+    @Size(min = 3, max = 10, message = "2")
+    private String city;
+
+    @Size(min = 3, max = 10, message = "3")
+    private String region;
+
+    @Valid
+    @NotNull(message = "Not null under valid")
```
Comment (SergeyPasko) [inline src/main/java/springData/dto/OfficeRequest.java]: if you use everything code - use code there too.  One of codes of internal  OfficeDetails validations

File: src/main/java/springData/exception/ControllerExceptionHandler.java [hunk 2]
```diff
@@ -35,4 +38,23 @@ ErrorMessage entityExistingProblem(Exception e) {
...
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public @ResponseBody ErrorMessage validationProblem(MethodArgumentNotValidException e){
+        LOG.warn("Request validation problem {}", e.getMessage());
+        FieldError fe = e.getBindingResult().getFieldError();
+        return new ErrorMessage(validationCodeDescription.get(fe.getDefaultMessage()));
+    }
+
+    @PostConstruct
+    private void intValidationCodeDescription(){
+        validationCodeDescription.put("1", "office number must not be greater then 500");
```
Comment (mkharkhu) [inline src/main/java/springData/exception/ControllerExceptionHandler.java]: why? you have a limit of at least 104?

File: src/main/java/springData/dto/OfficeRequest.java (hunk 1 above)
Comment (SergeyPasko) [reply src/main/java/springData/dto/OfficeRequest.java]: 4 or 5

Comment (SergeyPasko) [reply src/main/java/springData/dto/OfficeRequest.java]: In your case region, city, office can be null. Is ok for you?

File: src/main/java/springData/exception/ControllerExceptionHandler.java [hunk 3]
```diff
@@ -35,4 +38,23 @@ ErrorMessage entityExistingProblem(Exception e) {
...
+        validationCodeDescription.put("2", "we work only in 10 letters cities");
+        validationCodeDescription.put("3", "only 'S' letter regions required");
+        validationCodeDescription.put("4", "monthly sales must be greater then target 25000");
+        validationCodeDescription.put("5", "monthly target must be greater then 25000");
+        validationCodeDescription.put("6", "is it office details?");
```
Comment (mkharkhu) [inline src/main/java/springData/exception/ControllerExceptionHandler.java]: look carefully at the mistakes you return

File: src/test/java/springData/controller/OfficeControllerTest.java [hunk 4]
```diff
@@ -107,17 +107,14 @@ public void testUpdateOfficeByIdNotExist() throws Exception {
 
     @Test
     public void testDeleteOfficeByIdExist() throws Exception {
-        MvcResult mvcResult = mockMvc.perform(delete("/office/{id}", "21")).andDo(print()).andReturn();
+        MvcResult mvcResult = mockMvc.perform(delete("/office/{id}", "111111")).andDo(print()).andReturn();
         assertNotEquals(Status.OK.getStatusCode(), mvcResult.getResponse().getStatus());
     }
```
Comment (mkharkhu) [inline src/test/java/springData/controller/OfficeControllerTest.java]: test the insert)

File: src/main/java/springData/dto/OfficeRequest.java (hunk 1 above)
Comment (Artsapollo) [reply src/main/java/springData/dto/OfficeRequest.java]: Made changes but still have an issues with validation

**Pull request merged**